### PR TITLE
[fix] Fix flaky test avoidUseSameOpAddEntryBetweenDifferentLedger.

### PR DESCRIPTION
**Pull request by MarvinCai**
Fixes https://github.com/apache/pulsar/issues/17165

### Motivation
Fix flaky test avoidUseSameOpAddEntryBetweenDifferentLedger.

### Modifications
Test case avoidUseSameOpAddEntryBetweenDifferentLedger is trying to poll data from ledger.pendingAddEntries, but
when OpAddEntry is completed the callback will also poll data from ledger.pendingAddEntries, thus causing race condition.
Try to move to logic of creating new OpAddEntry when switched to a different ledger to a separate method so test won't
trigger actualy OpAddEntry compeltion and eliminating race condition for test.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: ( no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java [hunk 1]
```diff
@@ -1503,7 +1503,7 @@ public void operationComplete(Void v, Stat stat) {
                     currentLedgerEntries = 0;
                     currentLedgerSize = 0;
                     metadataMutex.unlock();
-                    updateLedgersIdsComplete(stat);
+                    updateLedgersIdsComplete();
```
Comment (eolivelli) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:1506]: This is not a fix for a flaky test, 
this is a change on ManagedLedgerImpl

I think that we should change the title

Comment (eolivelli) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:1506]: after re-reading the code I have verified that the behaviour did not change.
you only removed a useless parameter and split the method

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java [hunk 2]
```diff
@@ -1569,14 +1569,7 @@ private void updateLedgersListAfterRollover(MetaStoreCallback<Void> callback, Le
...
 
-    public synchronized void updateLedgersIdsComplete(Stat stat) {
-        STATE_UPDATER.set(this, State.LedgerOpened);
-        updateLastLedgerCreatedTimeAndScheduleRolloverTask();
-
-        if (log.isDebugEnabled()) {
-            log.debug("[{}] Resending {} pending messages", name, pendingAddEntries.size());
-        }
-
+    public void createNewOpAddEntryForNewLedger() {
```
Comment (eolivelli) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:1572]: can we add "VisibleForTesting" ?
this method is not meant to be called by other classes

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java (hunk 1 above)
Comment (eolivelli) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:1506]: so there is no need to change the title

**Review by eolivelli — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]
LGTM

**Pull request merged**